packages/indy-vdr/src/anoncreds/IndyVdrAnonCredsRegistry.ts

```diff
@@ -660,8 +854,171 @@ export class IndyVdrAnonCredsRegistry implements AnonCredsRegistry {
-    throw new AriesFrameworkError('Not implemented!')
+  public async registerRevocationStatusList(
+    agentContext: AgentContext,
+    { options, revocationStatusList }: IndyVdrRegisterRevocationStatusList
+  ): Promise<RegisterRevocationStatusListReturn> {
+    try {
+      // This will throw an error if trying to register a credential definition with a legacy indy identifier. We only support did:indy
+      // identifiers for registering, that will allow us to extract the namespace and means all stored records will use did:indy identifiers.
+      const { namespaceIdentifier, namespace } = parseIndyDid(revocationStatusList.issuerId)
+      const indyVdrPoolService = agentContext.dependencyManager.resolve(IndyVdrPoolService)
+      const pool = indyVdrPoolService.getPoolForNamespace(namespace)
+
+      agentContext.config.logger.debug(
+        `Registering revocation status list on ledger '${namespace}' with did '${revocationStatusList.issuerId}'`,
+        revocationStatusList
+      )
+
+      let writeRequest: CustomRequest
+
+      const { revocationRegistryDefinition } = await this.getRevocationRegistryDefinition(
+        agentContext,
+        revocationStatusList.revRegDefId
+      )
+
+      if (!revocationRegistryDefinition) {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve revocation registry definition '${revocationStatusList.revRegDefId}'`,
+          },
+        }
+      }
+
+      const { credentialDefinition } = await this.getCredentialDefinition(
+        agentContext,
+        revocationRegistryDefinition.credDefId
+      )
+
+      if (!credentialDefinition) {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve credential definition '${revocationRegistryDefinition.credDefId}'`,
+          },
+        }
+      }
+
+      const { schemaMetadata } = await this.getSchema(agentContext, credentialDefinition.schemaId)
+
+      if (!schemaMetadata?.indyLedgerSeqNo || typeof schemaMetadata.indyLedgerSeqNo !== 'number') {
+        return {
+          registrationMetadata: {},
+          revocationStatusListMetadata: {},
+          revocationStatusListState: {
+            revocationStatusList,
+            state: 'failed',
+            reason: `Unable to resolve schema '${credentialDefinition.schemaId}'`,
+          },
+        }
+      }
+
+      const schemaSeqNo = Number(schemaMetadata.indyLegderSeqNo)
+      const didIndyRevocationRegistryEntryId = getDidIndyRevocationRegistryEntryId(
+        namespace,
+        namespaceIdentifier,
+        schemaSeqNo,
+        credentialDefinition.tag,
+        revocationRegistryDefinition.tag
+      )
+
```

Comment:
The revocation registry definition id is already passed as part of the input to this method, so I don't think it's needed to fetch the schema + cred def + revoc reg to construct the id.

Rather we can take it from `revocationStatusList .revRegDefId`